### ISSUE TITLE
homepage: cannot follow any link

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#1676]
Seems like the homepage build went through: https://build.libelektra.org/jenkins/job/elektra-homepage/224/console

(Although there is an ssh error at the end?)

 It seems like a new homepage is online but it is completely broken: no link is working (you simply stay at the main page). Entering existing links leads to a white page.

@BernhardDenner Please fix, we want to release today but we cannot with this situation.

@Namoshek Do you have any idea?

**Comment by sanssecours** [comment]
> It seems like a new homepage is online but it is completely broken: no link is working (you simply stay at the main page).

Strange, the homepage seems to work correctly in my browser.

**Comment by BernhardDenner** [comment]
Yes, because I've reverted it TP the old Version again.

Am 31.10.2017 8:12 vorm. schrieb "René Schwaiger" <notifications@github.com

**Comment by markus2330** [comment]
Thank you for working on it, that is great!

**Comment by markus2330** [comment]
@BernhardDenner Thank you for working on it, that is great! Can you post in detail here what you changed, to help us ( @Namoshek ?) to locate the issue.

- Did nodejs change?
- Which rewrite rules do we have now?

A local build of the website displayed with grunt worked for me some days ago.

Is this one issue or several?

- [ ] Cannot follow any link on the web site.
- [ ] A blank page is shown when directly visiting a website.

When searching for this, I found suggestions to change something within Angular JS, but this seems ill-advised as it is working in other setups? (E.g. locally with grunt)

**Comment by BernhardDenner** [comment]
```
$nodejs -v
v6.11.4
$ npm -v
3.10.10
```

**Comment by BernhardDenner** [comment]
hmmm: https://stackoverflow.com/questions/41169385/http-get-success-is-not-a-function
package.json says: 

```
       "angular": "^1.5.8",
```
can this be related to that issue???

**Comment by markus2330** [comment]
I just tried with the same nodejs version: Only get some deprecated warnings about connect, minimatch, http2, but everything works as expected (locally with grunt). But when doing a new Elektra installation, it fails with a fsevents dependency problem.

On a complety new installed Debian system with nodejs 6.11.5, however, it works without problems.

So I would suggest to remove the npm cache and/or try upgrading to nodejs 6.11.5.

**Comment by Namoshek** [comment]
Yes, this may be the issue (although it seems weird because breaking changes should be prevented by the used semver syntax). Try removing the leading `^` and with some luck it works again. Possible reason is given here in [their changelog](https://github.com/angular/angular.js/blame/master/CHANGELOG.md#L488).

Alternative would be to keep the current npm setting, which is to use the most recent angular version (without breaking changes, so it would be `1.5.11` actually), and to change all calls in the code. Old syntax is `$http.get(...).success(function (data, status, headers, config) { ... }).error(function (data, status, headers, config) { ... });` vs. the new one `$http.get(...).then(function (response) { ... }, function (response) { ... });` (where the first function is the success and the second the error handler). The `response` object of the callbacks has all the properties that were given to the function in the old syntax directly (e.g. `response.status`, `response.data`, ...). An explanation is given in [their changelog](https://github.com/angular/angular.js/blob/master/CHANGELOG.md#http-due-to) as well.

I'm not sure if the security related fixes are worth updating and changing quite some calls...

**Comment by BernhardDenner** [comment]
Since I've started before your reply, I'm going the hard way of fixing all `http.get` calls.
Until now, I've manually fixed and updated most of them (hopefully). So please test and report any issues. The login isn't working right now, I'm already fixing that.

**Comment by markus2330** [comment]
Thanks for creating #1678. I'll leave it open a bit to give @Namoshek time to have a look at it.
